Examples/CaseStudies/SwiftUICaseStudies/04-HigherOrderReducers-FailureToast.swift

```diff
@@ -0,0 +1,145 @@
+import Combine
+import ComposableArchitecture
+import SwiftUI
+
+private let readMe = """
+  This screen demonstrates how the `Reducer` struct can be extended to enhance reducers with extra \
+  functionality.
+
+  In it we introduce an `errorHandling` reducer that consumes the failable `appReducer`. \
+  It handles the errors by way of a toast, and returns the usual non-failing reducer.
+
+  This form of reducer is useful if you want to centralize and handle failures in the same way. \
+  Without this, each routine executed in the `appReducer` would need to handle it's own failure.
+
+  Tapping the "Load Data" button will fail after one second and show a toast.
+  """
+
+enum AppError: Error, LocalizedError {
+  case api
+
+  var errorDescription: String? { "Whoops! There was a server error." }
+}
+
+struct AppState: Equatable {
+  var toastStatus: ToastStatus = .hiding
+  var data: [String] = []
+}
+
+enum ToastStatus: Equatable {
+  case showing(String)
+  case hiding
+
+  var isShowing: Bool {
+      switch self {
+      case .showing: return true
+      case .hiding: return false
+      }
+  }
+
+  var text: String {
+      switch self {
+      case .showing(let text): return text
+      case .hiding: return ""
+      }
+  }
+}
+
+enum AppAction {
+  case showToast(String)
+  case hideToast
```

Comment:
I was expecting a `case toastAction(ToastAction)` so the feature action knows less about the reusable toast functionality (I think the favorites example does something like this)

Reply:
Thank you - we've made the changes to make it more like the favourites example. 🙏